src/game.cpp

```diff
@@ -8917,13 +8937,29 @@ bool game::walk_move( const tripoint &dest_loc )
     }
 
     if( m.impassable( dest_loc ) && !pushing && !shifting_furniture ) {
+        if( vp_there && u.mounted_creature && u.mounted_creature.get()->has_flag( MF_RIDEABLE_MECH ) &&
+            vp_there->vehicle().handle_potential_theft( dynamic_cast<player &>( u ) ) ) {
+            auto mon = u.mounted_creature.get();
+            point diff = point( dest_loc.x - u.pos().x, dest_loc.y - u.pos().y );
```

Comment:
```suggestion
            tripoint diff = dest_loc - u.pos();
```
And than you can use it the same way later: `shove_vehicle( dest_loc + diff, ...`